### PR TITLE
beets.externalPlugins.check: init at 0.12.0 

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
I need this plugin

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @aszlig @domenkozar @pjones

### CODE REVIEW

**Comment by lovesegfault** [comment]
A test is still failing, I'm working on it

**Comment by lovesegfault** [comment]
All fixed.

**Review by aszlig — APPROVED** [review]

File: pkgs/tools/audio/beets/check-plugin.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  # patch out broken tests
+  patches = [ ./beet-check-tests.patch ];
+
+  prePatch =''
+    sed -i "/futures/d" setup.py
+  '';
+
+  checkPhase = "nosetests";
+
+  meta = {
```
Comment (worldofpeace) [inline pkgs/tools/audio/beets/check-plugin.nix]: can we do
```suggestion
  meta = with stdenv.lib; {
```

Comment (worldofpeace) [reply pkgs/tools/audio/beets/check-plugin.nix]: a maintainer would be cool too.

File: pkgs/tools/audio/beets/check-plugin.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  nativeBuildInputs = [ beets ];
+  checkInputs = [ pythonPackages.nose flac liboggz mp3val ];
+  propagatedBuildInputs = [ flac liboggz mp3val ];
+
+  # patch out broken tests
+  patches = [ ./beet-check-tests.patch ];
+
+  prePatch =''
+    sed -i "/futures/d" setup.py
```
Comment (worldofpeace) [inline pkgs/tools/audio/beets/check-plugin.nix]: why is futures not needed?

**Comment by worldofpeace** [comment]
This is completely impractical to test without
```diff
diff --git a/pkgs/tools/audio/beets/default.nix b/pkgs/tools/audio/beets/default.nix
index 3c2c474240f..b4dfa71ca96 100644
--- a/pkgs/tools/audio/beets/default.nix
+++ b/pkgs/tools/audio/beets/default.nix
@@ -250,6 +250,10 @@ in pythonPackages.buildPythonApplication rec {
 
   makeWrapperArgs = [ "--set GI_TYPELIB_PATH \"$GI_TYPELIB_PATH\"" "--set GST_PLUGIN_SYSTEM_PATH_1_0 \"$GST_PLUGIN_SYSTEM_PATH_1_0\"" ];
 
+  passthru = {
+    externalPlugins = plugins;
+  };
+
   meta = {
     description = "Music tagger and library organizer";
     homepage = http://beets.io;
```

Now I can actually build the external packages under ` beets.externalPlugins.check`, and your commit message can have this attribute.

File: pkgs/tools/audio/beets/check-plugin.nix (hunk 2 above)
Comment (lovesegfault) [reply pkgs/tools/audio/beets/check-plugin.nix]: Only needed for Python2, but I only intend on supporting Python3.

**Comment by lovesegfault** [comment]
@worldofpeace all fixed up :)

File: pkgs/tools/audio/beets/check-plugin.nix (hunk 2 above)
Comment (worldofpeace) [reply pkgs/tools/audio/beets/check-plugin.nix]: Can you comment this?

**Review by worldofpeace — APPROVED** [review]
LGTM, just maybe need a comment for the sed.

File: pkgs/tools/audio/beets/check-plugin.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  nativeBuildInputs = [ beets ];
+  checkInputs = [ pythonPackages.nose flac liboggz mp3val ];
+  propagatedBuildInputs = [ flac liboggz mp3val ];
+
+  # patch out broken tests
+  patches = [ ./beet-check-tests.patch ];
+
+  # patch out futures dependency, it is only needed for Python2 which we don't
+  # support.
+  prePatch =''
```
Comment (worldofpeace) [inline pkgs/tools/audio/beets/check-plugin.nix]:
```suggestion
  prePatch = ''
```

**Pull request merged**